src/arrays/quarray.jl

```diff
@@ -0,0 +1,148 @@
+import Base: size,
+    length,
+    getindex,
+    similar,
+    in,
+    ctranspose,
+    transpose,
+    summary,
+    zeros,
+    eye,
+    #TODO: Implement the below operations
+    *,.*,
+    /,./,
+    +,.+,
+    -,.-,
+    kron
+
+abstract AbstractQuArray{B<:(AbstractBasis...), T, N} <: AbstractArray{T,N}
+
+###########
+# QuArray #                 
+###########
+    # B<:(AbstractBasis...) allows us to have a basis for each dimension, 
+    # and gives us a less ambiguous way to determine if a QuArray will act
+    # like a vector, matrix, or tensor compared to just looking at the 
+    # dimension parameter N. 
+    type QuArray{B<:(AbstractBasis...), T, N, A} <: AbstractQuArray{B, T, N}
+        coeffs::A
+        bases::B
+        function QuArray(coeffs::AbstractArray{T, N}, bases::B) 
+            if checkbases(coeffs, bases) 
+                new(coeffs, bases)
+            else 
+                error("Coefficient array does not conform to input bases")
+            end
+        end
+    end
+
```

Comment:
I agree. I've implemented this now, let me know what you think. If there aren't any other issues (besides exportation, which we now have a separate issue for), I'm ready for this to be actually merged.